services/kafka/config.go

```diff
@@ -82,58 +84,59 @@ type WriteTarget struct {
 	PartitionAlgorithm string
 }
 
-func (c Config) WriterConfig(diagnostic Diagnostic, target WriteTarget) (kafka.WriterConfig, error) {
+func (c Config) writerConfig(diagnostic Diagnostic, target WriteTarget) (*kafka.Config, error) {
+	cfg := kafka.NewConfig()
+
 	if target.Topic == "" {
-		return kafka.WriterConfig{}, errors.New("topic must not be empty")
+		return cfg, errors.New("topic must not be empty")
 	}
-	var balancer kafka.Balancer
+	var partitioner kafka.PartitionerConstructor
 	if target.PartitionById {
 		switch target.PartitionAlgorithm {
-		case "crc32c", "":
-			balancer = &kafka.CRC32Balancer{}
+		case "crc32", "":
+			partitioner = func(topic string) kafka.Partitioner { return crcpartitioner{} }
 		case "murmur2":
-			balancer = &kafka.Murmur2Balancer{}
+			partitioner = func(topic string) kafka.Partitioner { return murmur2{} }
+		case "murmur3":
+			// because murmur3 is a new feature we don't need to do anything wild to keep compatibility
```

Comment:
would be better to describe the old wildness above, than call out this as being normal.